platform/ext/target/nordic_nrf/Native_Driver/spu.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "spu.h"
+#include <hal/nrf_spu.h>
+
+/* Platform-specific configuration */
+#define FLASH_SECURE_ATTRIBUTION_REGION_SIZE 0x8000
+#define SRAM_SECURE_ATTRIBUTION_REGION_SIZE  0x2000
+
+#define NUM_FLASH_SECURE_ATTRIBUTION_REGIONS 32
+#define NUM_SRAM_SECURE_ATTRIBUTION_REGIONS  32
```

Comment:
Yes.
Flash region size is 32 for nrf5340pdk and 16 for nrf5340dk (Erratum 19)
Number of RAM regions is double for nrf5340 vs nrf9160 since RAM size is double.

Reply:
I guess we do not care about the _dk_ with the Erratum19 @oyvindronningstad , can we assume we only care about the nRF5340pdk?

Reply:
It's the pdk that has the erratum (16 is correct, 32 is faulty). We definitely need to care about the dk when it comes.

Reply:
Anyways, I also thought about it during development, and had decided to ignore this for now, but since you both bring this up, maybe let's fix this.
For that we need #ifdef 's here, though. @oyvindronningstad which are or shall be the directives here?

Reply:
btw, i am also thinking these should be in some internal header, not in spu.c, any ideas?

Reply:
We've got `SPU_FLASH_REGION_SIZE` in `region_defs.h`. We could add a corresponding definition for RAM regions. And number of regions can be calculated from the total size of the respective memory, can't it?

Reply:
> We've got `SPU_FLASH_REGION_SIZE` in `region_defs.h`. We could add a corresponding definition for RAM regions. And number of regions can be calculated from the total size of the respective memory, can't it?

If you know the region size, yes.

Reply:
Alright, I am adding the SPU_SRAM_REGION_SIZE for now in region_defs.h.
I am also calculating the number of regions as proposed.

Let's not bother, yet, about _pdk and _dk @oyvindronningstad , this should come once we have different cmake targets for these, or until we have a way to detect the DK version in runtime.

For the first prototype this should not be a blocker imho.